### DISCUSSION TITLE
Better tag search

### DISCUSSION THREAD

**Discussion by server-imp** [jellyfin/jellyfin#11550] [General]
With the release of 10.9, searching has become leaps and bounds better by allowing you to search for tags, however I am a little dissapointed that you cannot search for multiple tags at once. At least I haven't been able to find a way how to do it.

For example, if I had a movie in my library containing these two tags: "heist", "desert", then you would expect it to show up when you search for "desert heist" or "heist desert", or "heist,desert" etc but currently when you provide multiple tags it doesn't generate any results at all.

My suggestion is to provide a way of searching tags where it generates results that all contain the provided tags, even if it would require a special syntax such as "heist,desert" or "heist|desert" for example.

On a related note - I think that if you use the filter in the library and select multiple categories, for example "action" and "comedy", it should only show items that have both of these categories, not everything that has action and comedy seperately.

**Comment by bradbeattie** [comment]
Hey hey. I implemented the tag searching in 10.9, so happy to hear you're enjoying it. With respect to its deficienciee, I suspect the best route forward would be to

1. Adjust the search functionality to permit new search plugins.
2. Write a new plugin that connects Jellyfin to an instance of Meilesearch or something like that
3. If step 2 proves relatively simple, powerful, and popular, advocate for integrating it as a "default if present, falling back to old search if not".

The alternative is to try to augment the existing search functionality which is never going to be as full featured as a dedicated search app (e.g. multiple term searching, correcting misspellings, etc).

**Comment by madhatr** [comment]
Is there a list of tags?  I'm trying to search for Sci-Fi movies and I get very different results.
0 results for Sci-Fi
Some results for science
some results for science fiction

There is some overlap but not much and i have way more than some sci-fi on my server.

Do I need a third party app to tag them or something?

**Reply by bbeattie-phxlabs** [reply]
Tags should appear in item details (look below the title and above the genre). Tags should be automatically populated presuming your content links up to external content IDs (e.g. TheMovieDb ID).
